cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -112,6 +112,9 @@ struct ARROW_EXPORT AzureOptions {
   /// This will be ignored if non-empty metadata is passed to OpenOutputStream.
   std::shared_ptr<const KeyValueMetadata> default_metadata;
 
+  /// Whether OutputStream writes will be issued in the background, without blocking.
+  bool background_writes = false;
```

Comment:
I don't feel strongly, but personally I think we should stop adding functionality and perhaps remove some from  `ExtractFromUriQuery()` as was suggested in https://github.com/apache/arrow/issues/43197

Reply:
If we can specify an option value by URI something like `abfs://.../...?background_writes=yes`, we don't need to implement `AzureOptions` bindings. For example, Apache Arrow C GLib doesn't provide `S3Options` nor `GCSOptions` but it can use `S3FileSystem` and `GCSFileSystem`.
I think that #43197 is for security. But this is not related to security.